### PR TITLE
Herokuize

### PR DESCRIPTION
**Pull request by dferrantelli**
This pull request includes a

- [ ] Bug fix
- [X] New feature
- [ ] Translation

The following changes were made
Added the option to set options via environment variable, added necessary files for this to run in heroku (or a heroku-like platform). Please excuse my horrible commits, trying to rebase lead to so many merge conflicts..
-
-
-

If this is related to an existing ticket, include a link to it as well.

### CODE REVIEW

**Comment by betrisey** [comment]
I had to add the host parameters to get it working
python example.py -H 0.0.0.0 -P $PORT

According to the logs, the requests come from another host
`2016-07-19T07:54:36.755401+00:00 app[web.1]: 10.86.165.146 - - [19/Jul/2016 07:54:36] "GET / HTTP/1.1" 200 -`

**Comment by dferrantelli** [comment]
@betrisey fixed - I had set that in my env, that's why i didn't notice it

**Comment by DanielVolz** [comment]
@dferrantelli could you please make a short guide on how to configure it with heroku?

**Comment by dferrantelli** [comment]
@DanielVolz Added! Even included a deploy button to make things super easy!

**Comment by DanielVolz** [comment]
@dferrantelli <3 you're the best!! Thank you very much!!!

**Comment by dferrantelli** [comment]
@AHAAAAAAA with almost every new feature that get's added i need to fix merge issues, any chance this can go in soon?

**Comment by Chlodochar** [comment]
@dferrantelli That's probably because the commits are touching a large portion of the application. It should be possible to Heroku-ize it all with just an `Procfile` (and optionally app.json). Pass all environment variables as parameters and you're done.

In addition, it's best to clean up the commits that are to be merged. This PR currently has 18 commits. Squash those commits to a single one and it'll be easier to merge as well.

File: README.md [hunk 1]
```diff
@@ -12,3 +12,10 @@ Building off [Mila432](https://github.com/Mila432/Pokemon_Go_API)'s PokemonGo AP
 
 ---
 For instructions, please refer to [the wiki](https://github.com/AHAAAAAAA/PokemonGo-Map/wiki)
+
+# Heroku
+Host your own Rocket.Chat server for **FREE** with [One-Click Deploy](https://heroku.com/deploy)
```
Comment (fosron) [inline README.md]: Rocket.chat ?

**Comment by MikeDX** [comment]
Team Rocket Chat?

File: README.md (hunk 1 above)
Comment (MikeDX) [inline README.md]: Team Rocket Chat? :D

**Comment by dferrantelli** [comment]
@Chlodochar i can try, but there are going to be a lot of conflicts. Github has a squash and merge option :) Also, just a proc file isn't enough, as the app doesn't currently look for env vars, hence all my changes to the argparse section, and also why every new feature means I need to fix merge conflicts.

**Comment by dferrantelli** [comment]
I just realized what you meant about the env vars - that doesn't work if you don't set say an only list, but the only list param is expecting an argument, so it would fail to start (I tried that option first)

**Comment by Danm72** [comment]
I've been using it on Heroku all day with only a Procfile.

web: python example.py -u $USERNAME -p $PASSWORD --port $PORT --host 0.0.0.0 -l "$LOCATION" -i "$IGNORES" -st $STEPS -ar $REFRESH

screen shot 2016-07-19 at 11 42 56 p m

**Comment by Danm72** [comment]
Great call on the deploy button though!

**Comment by dferrantelli** [comment]
@Danm72 have you tried putting -o $ONLY and not setting the $ONLY variable? that is what i'm referring to. if you set an argument in the procfile that requires an argument, but the env var doesn't exist, the app will fail to start because you are not providing the arg. In order for the procfile to work for everyone, we would need to put every option in the procfile, creating many failure points.

**Comment by Danm72** [comment]
Ah, I see what you mean that is a valid concern. Would you consider putting them into two separate PRs if it'll help the guys merge the deploy/procfile etc quicker?

**Comment by dferrantelli** [comment]
squashed my commits in a different branch using the git pr squash and merge option (doing it manually involved solving merge conflicts that I did not want to touch). It is now one clean commit that will merge if no more existing options change. If the maintainers want two prs for the procfile and env stuff i can split it out. further convo should happen in #449

**Pull request closed without merging**